model/src/test/kotlin/utils/LicenseResolverTest.kt

```diff
@@ -0,0 +1,244 @@
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.ScanResultContainer
+import org.ossreviewtoolkit.model.ScanSummary
+import org.ossreviewtoolkit.model.ScannerDetails
+import org.ossreviewtoolkit.model.ScannerRun
+import org.ossreviewtoolkit.model.TextLocation
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.ScannerConfiguration
+
+import java.time.Instant
+import java.util.concurrent.atomic.AtomicInteger
+
+private fun OrtResult.addPackage(pkg: Package): OrtResult = copy(
+    analyzer = analyzer!!.copy(
+        result = analyzer!!.result.copy(
+            packages = (analyzer!!.result.packages + pkg.toCuratedPackage()).toSortedSet()
+        )
+    )
+)
+
+private fun OrtResult.addScanResult(id: Identifier, scanResult: ScanResult): OrtResult {
+    val scanResults = scanner!!.results.scanResults
+    var scanResultsForId = scanResults.find { it.id == id }?.let {
+        scanResults.remove(it)
+        it.results
+    } ?: emptyList()
+
+    scanResults += ScanResultContainer(id, scanResultsForId + scanResult)
+
+    return copy(
+        scanner = scanner!!.copy(
+            results = scanner!!.results.copy(
+                scanResults = scanResults
+            )
+        )
+    )
+}
+
+private fun OrtResult.getVcsProvenance(id: Identifier): Provenance? =
+    getProject(id)?.let { Provenance(vcsInfo = it.vcsProcessed) }
+        ?: getPackage(id)?.let { Provenance(vcsInfo = it.pkg.vcsProcessed) }
+
+private fun OrtResult.getSourceArtifactProvenance(id: Identifier): Provenance? =
+    getPackage(id)?.let { Provenance(sourceArtifact = it.pkg.sourceArtifact) }
+
+private fun createEmptyOrtResult() = OrtResult(
+    Repository.EMPTY,
+    AnalyzerRun(
+        environment = Environment(),
+        config = AnalyzerConfiguration(ignoreToolVersions = true, allowDynamicVersions = true),
+        result = AnalyzerResult.EMPTY
+    ),
+    ScannerRun(
+        environment = Environment(),
+        config = ScannerConfiguration(),
+        results = ScanRecord(
+            scanResults = sortedSetOf(),
+            storageStats = AccessStatistics()
+        )
+    )
+)
+
+class LicenseResolverTest : WordSpec() {
+    private val nextId = AtomicInteger(1)
+    private lateinit var ortResult: OrtResult
+    private lateinit var packageConfigurationProvider: PackageConfigurationProvider
+
+    private fun createResolver() = LicenseResolver(ortResult, packageConfigurationProvider)
+
+    private enum class ProvenanceType { VCS, SOURCE_ARTIFACT }
+
+    private fun setupScanResult(id: Identifier, type: ProvenanceType, licenseFindings: Collection<LicenseFinding>) {
+        val scanResult = ScanResult(
+            provenance = when (type) {
+                ProvenanceType.SOURCE_ARTIFACT -> ortResult.getVcsProvenance(id)
+                else -> ortResult.getSourceArtifactProvenance(id)
```

Comment:
Is this really correct? I would have expected that `getSourceArtifactProvenance()` is used for `ProvenanceType.SOURCE_ARTIFACT`.

Reply:
Nope it's wrong, good catch!